kernel/mempool.c

```diff
@@ -146,8 +147,13 @@ void *k_mem_pool_malloc(struct k_mem_pool *pool, size_t size)
 		return NULL;
 	}
 
+	/* MISRA-C 21.15 requires the pointer arguments for memcpy be
+	 * compatible types. Unfortunately just cast block.id inside the function
+	 is not enough. */
```

Comment:
The block comment is not correctly formatted here.
Should be
```
/*
 * xxxx
 * xxxx
 */
```
Could you elaborate why cast is not enough here and also add that info to the comment?

Reply:
Jumping in for a shameless digression. 

That requirement that the comment end token be on a line by itself comes from checkpatch.  It's not a particularly standard thing outside the Linux world, where that kind of formatting is if anything far more common (e.g. in emacs, if you flow a C comment, it puts the terminator on the last line and I need to fix these up all the time).

Since we're getting into the world of code formatting anyway with all the MISRA stuff... who do I talk to to start the giant bike shed about comment formatting and getting this format approved?

Ditto for double-slash EOL comments, which have been part of the C standard for nearly two decades now yet are still disallowed out of blind inertia.

Reply:
@andyross mailing list RFC I guess, these seem like reasonable changes to the coding standard.

Reply:
Two things C has had for decades now that seem to be resisted for no real apparent reason: "//" comments, and declaring variables inline (such as where they first have a value).